style_guides/pl_sql_naming_conventions.md

```diff
@@ -0,0 +1,380 @@
+    
+    -- Private function, only available inside the package
+    FUNCTION fw_get_private_value
+        RETURN VARCHAR2
+    IS
+    BEGIN
+        RETURN 'A Private Value';
+    END fw_get_private_value;
+END pkw_example_package;
+/
+```
+
+
+
+## Tables
+
+### Table Names
+
+Tables should be given a meaningful name that describes the purpose of the table. All table names should be plural and where the table name contains more than one word only the last should be plural. 
+
+Table names should be underscore separated. 
+
+Tables that are related to a single project or development should be prefixed with an identifier. 
+
+For example:
+
+```
+timetable_programmes
+timetable_programme_years
+timetable_programme_year_modules
+timetable_export_log
+```
+
+**When creating new tables ensure that you get the right balance of brevity and meaningfulness**. Don't create tables with unnecessarily long names or names that are so short that it is difficult to understand their purpose. 
+
+
+### Constraints
+
+Table constraints should be named as followed. 
+
+```
+pk_<table>                 -- Primary Key
+fk_<table>_<optional_text> -- Foreign Key
+```
+
+### Table comments
+
+Table and column comments should be included in the creation script for all new tables. 
+
+```sql
+COMMENT ON TABLE schema.table IS 'Overall table description';
+
+
+COMMENT ON COLUMN schema.table.column1 IS 'Column 1 description';
+COMMENT ON COLUMN schema.table.column2 IS 'Column 2 description';
+```
+
+### Temporary tables
+
+The name for temporary tables should always link back to the Service Now incident or JiRA issue number that they have been created for, for example *srs_001_swradms_table_backup* . 
+
+Note that table names cannot contain hyphen ( - ) characters and these should be replaced with an underscore. 
+
+**You should always plan and schedule change requests to remove these tables when no longer unnecessary. Your JiRA issue should not be marked as done until all temporary tables have been removed.**
+
+### File Names
+
+Scripts that create or update tables should have filenames in the following format, for example ssdt-001_table.sql
+
+```
+ssdt-001_<table_name>.sql        -- Create table script
+ssdt-001_alter_<table_name>.sql  -- Alter table script
+```
+
+## Triggers
+
+### Trigger Names
+
+Triggers should be given a meaningful name that indicates the triggering event (UPDATE/DELETE/INSERT), timing (BEFORE/AFTER) and type ( COMPOUND TRIGGER? ) in the following format:
+
+```
+tw_<COMPOUND>_<timing>_<event>_on_<table> -- <COMPOUND> is optional and event can be duplicated to indicate multiple triggering events
+```
+
+Use the following abbreviations for timing:
+
+```
+AFT -- After
+BEF -- Before
+```
+
+Use the following abbreviations for triggering events:
+
+``` 
+UPD -- Update
+INS -- Insert
+DEL -- Delete
+```
+
+For example:
+
+```
+TW_AFT_UPD_ON_SWBDEGC
+TW_BEF_DEL_ON_SWRADMS
+TW_AFT_INS_UPD_ON_STVDEPT
+TW_COMPOUND_AFT_INS_ON_SCBCRSE
+```
```

Comment:
What is the issue with using the full words here rather than abbreviating them?

Reply:
its one of those things that when you look at it we are saving at max 6 characters from a name so seems fine to just have the whole words